lambdas/src/services/security/jwe-decrypter.ts

```diff
@@ -61,18 +71,80 @@ export class JweDecrypter {
         return plainText;
     }
 
-    private async getKey(encryptedKey: string): Promise<Uint8Array | undefined> {
+    private async getDecryptedCek(encryptedKey: string): Promise<Uint8Array | undefined> {
+        const cipherTextBlob = base64url.decode(encryptedKey);
+        if (process.env.ENV_VAR_FEATURE_FLAG_KEY_ROTATION === "true") {
+            return await this.decryptWithKeyRotation(cipherTextBlob);
+        }
+        return await this.decryptWithLegacyKey(cipherTextBlob);
+    }
+
+    private async decryptWithKeyRotation(cipherTextBlob: Uint8Array): Promise<Uint8Array | undefined> {
+        logger.info("Key rotation enabled. Attempting to decrypt with key aliases.");
+        const decryptedWithAlias = await this.decryptWithAliases(cipherTextBlob);
+        if (decryptedWithAlias) {
+            return decryptedWithAlias;
+        }
+        if (process.env.ENV_VAR_FEATURE_FLAG_KEY_ROTATION_LEGACY_KEY_FALLBACK === "true") {
+            logger.warn("Failed to decrypt with all available key aliases, falling back to legacy key.");
+            return await this.decryptWithLegacyKey(cipherTextBlob);
+        }
+        throw new Error("Failed to decrypt with all available key aliases.");
+    }
+
+    private async decryptWithAliases(cipherTextBlob: Uint8Array): Promise<Uint8Array | undefined> {
```

Comment:
Sorry if I'm missing something but this method just seems to return a key rather than do a decryption?